src/mixed-line-bar-chart/interfaces.ts

```diff
@@ -73,15 +73,27 @@ export namespace MixedLineBarChartProps {
     type: 'line';
   }
 
-  export interface ThresholdSeries extends Omit<IDataSeries<never>, 'data' | 'valueFormatter'> {
+  export interface ThresholdSeries<T = any> extends Omit<IDataSeries<never>, 'data' | 'valueFormatter'> {
```

Comment:
Ideally to have here:

```
export type ThresholdSeries<T> = YThresholdSeries<never> | XThresholdSeries<T>;
```

However, this way TS can't figure out always which one is to choose.

Reply:
Shall we discuss it [here](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgBoBUAWUIGdMD2ANgCbIDeAsAFDJ3JgCeADhAFzIDkY2ehpnADQ16yAB4cQAVwC2AI2gBuEfUYB+SRABuSmgF8ah6qEixEKAJpYc+YmSq16TVh269bA4Y7qNJshVDK3uIayCDautQG1DTOKNZ8dsgAvGgJHmQAPshW7vwkQTQIBCC4YAx5dhzp+SkUKnRxrjw2+UIN4hwADF56QA)?

Reply:
In the TS playground it seems to work.

```
type ChartDataTypes = number | string | Date;

interface XThreshold<T extends ChartDataTypes> {
    type: 'threshold',
    x: T;
    y?: never;
}


interface YThreshold {
    type: 'threshold',
    y: number;
    x?: never;
    valueFormatter?: (value: number) => string
}

type Threshold<T extends ChartDataTypes> = XThreshold<T> | YThreshold;

const threshold = {
    type: 'threshold',
    y: 0,
} as const;

function test<T extends ChartDataTypes>(threshold: Threshold<T>) {
    // implementation
}

test({ type: 'threshold', x: '1' })
test({ type: 'threshold', y: 11 })
test(threshold)
```